### PR TITLE
require explicit `key` query param for registrant lookup using discord/osu user IDs

### PR DESCRIPTION
**Pull request by ILW8**

### CODE REVIEW

File: discord/views.py [hunk 1]
```diff
@@ -45,22 +45,51 @@ class TournamentPlayerViewSet(viewsets.ModelViewSet):
...
+
+        if lookup_type == "osu":
+            try:
+                if obj is None:
+                    obj = queryset.get(osu_user_id=self.kwargs[lookup_url_kwarg])
+            except queryset.model.DoesNotExist:
+                raise error_404
+
+        if obj is None:
+            raise error_404
```
Comment (myssto) [inline discord/views.py]: Lines 71-91 could probably get cleaned up a lot with a match case
```py
match lookup_type
    case "discord":
        pass
    case "osu":
        pass
    case _:
        raise error_404
```
other than that this looks ok

File: discord/views.py [hunk 2]
```diff
@@ -45,22 +45,51 @@ class TournamentPlayerViewSet(viewsets.ModelViewSet):
...
+        lookup_url_kwarg = self.lookup_url_kwarg or self.lookup_field
+
         try:
-            return super().get_object()
+            if lookup_type is None or lookup_type in ('id', 'pk'):
+                return super().get_object()
         except Http404 as pk404:
-            lookup_url_kwarg = self.lookup_url_kwarg or self.lookup_field
-            if lookup_url_kwarg not in self.kwargs:
+            if lookup_url_kwarg not in self.kwargs:  # I'm no longer sure what purpose this really serves...
```
Comment (myssto) [inline discord/views.py:65]: Truthfully I'm not sure either... I'm pretty sure I grabbed it from `super().get_object()` when I made my change

**Review by myssto — APPROVED** [review]
Could be cleaned up a little but looks good

File: discord/views.py (hunk 1 above)
Comment (ILW8) [reply discord/views.py]: oh yeah, structural matching is a thing in python now... I did agonize over how to implement this in a cleaner way, and IIRC it was also 5 in the morning

**Pull request merged**